src/examples/nupic/algorithms/HelloSP_TP.cpp

```diff
@@ -0,0 +1,87 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2013-2014, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ---------------------------------------------------------------------
+ */
+
+#include <iostream>
+#include <vector>
+#include <algorithm>    // std::generate
+#include <ctime>        // std::time
+#include <cstdlib>      // std::rand, std::srand
+#include <cmath> 	// pow
+
+#include "nupic/algorithms/SpatialPooler.hpp"
+#include "nupic/algorithms/Cells4.hpp"
+
+using namespace std;
+using namespace nupic;
+using namespace nupic::algorithms::spatial_pooler;
+using namespace nupic::algorithms::Cells4;
+
+// function generator:
+int RandomNumber01 () { return (rand()%2); } // returns random (binary) numbers from {0,1}
+
+int main(int argc, const char * argv[])
+{
+const int DIM = 2048; // number of columns in SP, TP
+const int DIM_INPUT = 10000;
+const int TP_CELLS_PER_COL = 10; // cells per column in TP
+const int EPOCHS = pow(10, 4); // number of iterations (calls to SP/TP compute() )
+
+  vector<UInt> inputDim = {DIM_INPUT};
+  vector<UInt> colDim = {DIM};
+
+  // generate random input
+  vector<UInt> input(DIM_INPUT);
+  vector<UInt> outSP(DIM); // active array, output of SP/TP
+  const int _CELLS = DIM * TP_CELLS_PER_COL;
+  vector<UInt> outTP(_CELLS);   
+  Real rIn[DIM] = {}; // input for TP (must be Reals)
+  Real rOut[_CELLS] = {};
+
+  // initialize SP, TP
+  SpatialPooler sp(inputDim, colDim);
+  Cells4 tp(DIM, TP_CELLS_PER_COL, 12, 8, 15, 5, .5, .8, 1.0, .1, .1, 0.0, false, 42, true, false);
```

Comment:
`tp` isn't the right name. `tm` would be better (for "temporal memory") or just `cells` since it isn't actually a full implementation of the temporal memory.